qemu/tests/win_virtio_serial_data_transfer.py

```diff
@@ -0,0 +1,203 @@
+                       "data is:\n%s Guest receive data:\n%s" % (
+                        ori_data, output))
+                raise error.TestFail(err)
+            self.event = 'received'
+        finally:
+            data_file = "c:\\%s" % os.path.basename(data_file)
+            session.cmd("del /F %s" % data_file)
+            session.close()
+
+    def _get_session(self):
+        """ return guest shell session"""
+        self.vm.verify_alive()
+        login_timeout = int(self.params.get("login_timeout", 360))
+        return self.vm.wait_for_login(timeout=login_timeout)
+
+    @error.context_aware
+    def install_required_scripts(self):
+        """
+        install required script into guest which will use to
+        receive data from host
+        """
+        deps_dir = data_dir.get_deps_dir("win_serial")
+        error.context("Copy test scripts to guest.", logging.info)
+        guest_scripts = self.params["guest_scripts"]
+        for script in guest_scripts.split(";"):
+            src_file = os.path.join(deps_dir, script)
+            self.vm.copy_files_to(src_file, "c:\\", timeout=60)
+        self.event = "prepare"
+
+    @error.context_aware
+    def uninstall_required_scripts(self):
+        """ cleanup installed script in guest """
+        guest_scripts = self.params["guest_scripts"]
+        error.context("Copy test scripts to guest.", logging.info)
+        session = self._get_session()
+        try:
+            for script in guest_scripts.split(";"):
+                script = "c:\\%s" % script
+                session.cmd("del /F %s" % script, timeout=60)
+            self.event = "cleanup"
+        except Exception:
+            logging.warn("Failed to remove %s" % guest_scripts)
+        finally:
+            session.close()
+
+
+def run(test, params, env):
+    """
+    QEMU 'Windows virtio-serial data transfer' test
+
+    1) Start guest with one virtio-serial-pci and two virtio-serial-port.
+    2) Make sure vioser.sys verifier enabled in guest.
+    3) Transfering data from host to guest via virtio-serial-port in a loop.
+    4) Reboot guest.
+    5) Repeat step 3.
+    6) Reboot guest by system_reset qmp command.
+
+    :param test: QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+    def logtime_transfer(win_serial_test, data_file, repeat):
+        """
+        Run data transfer from host to guest in loop
+        """
+        for count in xrange(1, repeat + 1):
+            logging.info("transfer repeat %d times" % count)
+            win_serial_test.transfer_data(data_file)
+
+    transfer_timeout = int(params.get("transfer_timeout", 900))
+    repeat = int(params.get("repeat_times", 1))
+    deps_dir = data_dir.get_deps_dir("win_serial")
+    send_script = params.get("host_send_script", "serial-host-send.py")
+    data_file = os.path.join(deps_dir, send_script)
+    interrupt = params.get("interrupt", "yes") == "yes"
```

Comment:
It could reboot/shutdown successfully during data transfer on win10-32bit after applying the PR.
cml:python ConfigTest.py --testcase=win_virtio_serial_data_transfer.interrupt.with_reboot --guestname=Win10 --platform=i386 --clone=no
result:
(1/1) smp_8.16384m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win10.i386.io-github-autotest-qemu.win_virtio_serial_data_transfer.interrupt.with_reboot: PASS (2338.40 s)

Acked-by: Yu Wang wyu@redhat.com